src/portable/dialog/da146xx/dcd_da146xx.c

```diff
@@ -363,13 +402,14 @@ static void start_rx_packet(xfer_ctl_t *xfer)
   xfer->last_packet_size = 0;
   if (xfer->max_packet_size > FIFO_SIZE && remaining > FIFO_SIZE)
   {
-    if (try_allocate_dma(epnum, TUSB_DIR_OUT))
+    if (try_allocate_dma(epnum, TUSB_DIR_OUT) && !xfer->ff)
```

Comment:
Conditions should probably be switched, there is no need to allocate dma if it will not be used due to second condition.
Same applies to line 452 below.